File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map; //establishing an hash map
        int length = nums.size();
        for( int i = 0; i < length; i++){
            map[nums[i]] = i; //putting values on the hashmap
        }

        for (int i = 0; i < length; i++){
            int diffr = target - nums[i];
            if (map.find(diffr) != map.end() && map[diffr] != i){
                return {i, map[diffr]};//checking on the hashmap
            }
        }

        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i in range(len(nums)):
            seen[nums[i]] = i

        for i in range(len(nums)):
            y = target - nums[i]

            if y in seen and seen[y]!= i:
                return [i, seen[y]]

